src/test/java/com/comcast/zucchini/FastrunZucchiniCukesTest.java

```diff
@@ -105,12 +105,10 @@ public void cleanup(TestContext out) {
     public Iterator<CucumberFeatureHolder> fastrunIteratorFactory(Iterator<CucumberFeature> iterator) {
         Iterator<CucumberFeatureHolder> response = super.fastrunIteratorFactory(iterator);
 
-        if (null == singleton) {
-            synchronized (this) {
-                if (null == singleton) {
-                    singleton = response;
-                    wrapped = new TestIterator<CucumberFeatureHolder>(response);
-                }
+        synchronized (this) {
```

Comment:
This test was failing in some weird timing way, traced it down to this null check... not really sure why it was there.

Reply:
Looks like this is a broken version of https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java